### PR TITLE
feat/refactor(pkg/log): add log writer stream

### PR DESCRIPTION
**Pull request by yqlbu**
### Background

Use `MultiLevelWriter` instead of `MonoLevelWriter` for `ConsoleWriter` and `FileWriter`.

> **Note**
> By default, the app will NOT write logs to a file unless `--log-file` flag is passed.

### References

https://github.com/natefinch/lumberjack

### Configs

systemd service

```conf
[Unit]
Description=juicity Service
Documentation=https://github.com/juicity/juicity
After=network.target nss-lookup.target

[Service]
Type=simple
User=root
ExecStart=/usr/bin/juicity-server run -c /etc/v2ray-agent/juicity/conf/config.json -f /var/log/juicity/juicity.log
Restart=on-failure
LimitNPROC=512
LimitNOFILE=infinity

[Install]
WantedBy=multi-user.target
```

### Checklist

- [x] The Pull Request has been fully tested
- [ ] There's an entry in the CHANGELOGS
- [x] There is a user-facing docs PR against https://github.com/daeuniverse/dae

### Full changelogs

- chore(deps): add lumberjack (log writer)
- feat(pkg/log): add log writer stream
- fixture(mock): add unit test
- refactor/feat: add support for multiple writer streams
- fixture(mock): add test case with ConsoleWriter(Stdout.noColor)
- feat(cmd/{client,server}): add json-log and no-color flags

### Issue reference

NA

### Test Result

Unit Test:

```bash
go test -v ./pkg/log/
```

image

image

VPS:

image

### CODE REVIEW

**Comment by yqlbu** [comment]
Also, we may consider persisting constants for `log` package in `/internal/constant/log.go`

**Comment by yqlbu** [comment]
Still need to work on the documentation side. However, functionality-wise it is considered as finished.

**Comment by yqlbu** [comment]
> Still need to work on the documentation side. However, functionality-wise it is considered as finished.

Resolved in https://github.com/juicity/juicity/pull/39/commits/0f5850f202a1d9f4f595c54f8af78c4bdbe542c5

**Comment by yqlbu** [comment]
Ready for review.

File: cmd/client/README.md [hunk 1]
```diff
@@ -26,18 +26,26 @@ make juicity-client
...
+  "log_level": "info"
 }
 ```
 
 - `listen` is the address that the socks5 and http server listen at.
 - Optional values of `congestion_control`: cubic, bbr, new_reno.
 - `sni` can be omitted if domain is given in `server`.
 - Set environment variable `QUIC_GO_ENABLE_GSO=true` to enable GSO, which can greatly improve the performance of sending and receiving packets. Notice that this option needs the support of NIC features. See more: <https://github.com/juicity/juicity/discussions/42>
+
+## Run Options
```
Comment (mzz2017) [inline cmd/client/README.md:45]:
```suggestion
## Arguments
```

File: cmd/client/README.md [hunk 2]
```diff
@@ -26,18 +26,26 @@ make juicity-client
...
+
+- `--config`: specify config file path
+- `--disable-timestamp`: disable timestamp
+- `--log-file`: write logs to file
+- `--no-log-color`: disable colorful log output
+- `--json-log-format`: use json log format
```
Comment (mzz2017) [inline cmd/client/README.md:51]: --log-file
--log-disable-color
--log-format (optional values: raw, json)

Comment (yqlbu) [reply cmd/client/README.md:51]: Cool, will fix it in a sec.

**Comment by yqlbu** [comment]
## Test Result

image

File: cmd/client/README.md (hunk 2 above)
Comment (yqlbu) [reply cmd/client/README.md]: Resolved in https://github.com/juicity/juicity/pull/39/commits/309a62b2fbc85589e6e43c3ce3054678305fadcf

**Review by mzz2017 — APPROVED** [review]
LGTM. Good Job!

**Pull request merged**